### PR TITLE
rgw: fix stats for versioned buckets after reshard

### PR DESCRIPTION
**Pull request by ivancich**
When a versioned bucket is resharded, the stats for bucket index entries of type PlainIdx and InstanceIdx were both accumulated to determine the bucket stats. This caused a doubling of some stats, such as bytes used. This fix makes certain that only PlainIdx entries are accumulated. This works for unversioned buckets as well, as they only have Plain Idx entries.

Also, do a little cleanup by converting enum BIIndexType to an enum class.

Tracker: https://tracker.ceph.com/issues/37473

### CODE REVIEW

**Comment by ivancich** [comment]
The solution is quite short spanning lines 250-260'ish in cls_rgw_types.cc . The rest is converting an enum to an enum class.

File: src/cls/rgw/cls_rgw_types.cc [hunk 1]
```diff
@@ -247,6 +247,8 @@ bool rgw_cls_bi_entry::get_info(cls_rgw_obj_key *key, uint8_t *category, rgw_buc
   using ceph::decode;
   switch (type) {
     case PlainIdx:
+        account = true;
```
Comment (mattbenjamin) [inline src/cls/rgw/cls_rgw_types.cc]: ok

File: src/cls/rgw/cls_rgw.cc [hunk 2]
```diff
@@ -2225,13 +2225,13 @@ static int rgw_bi_get_op(cls_method_context_t hctx, bufferlist *in, bufferlist *
   string idx;
 
   switch (op.type) {
-    case PlainIdx:
+    case BIIndexType::Plain:
```
Comment (mattbenjamin) [inline src/cls/rgw/cls_rgw.cc]: I do like enum class, seems harmless to backport

**Comment by ivancich** [comment]
jenkins retest this please

**Pull request merged**